source/darwin/nw_socket.c

```diff
@@ -1255,27 +1658,117 @@ static void s_handle_write_fn(
     aws_event_loop_schedule_task_now(nw_socket->event_loop, &args->task);
 }
 
+/*
+ * Because TLS negotiation is handled by Apple Network Framework connection using its parameters, we need access to a
+ * number of items typically not needed until the TLS slot and handler are being initialized. This function along with
+ * aws_socket_retrieve_tls_options_fn() are used to gain access to those items.
+ */
+static int s_setup_tls_options_from_context(
+    struct nw_socket *nw_socket,
+    struct aws_tls_connection_context *tls_connection_context) {
+    /* The host name is needed during the setup of the verification block */
+    if (tls_connection_context->host_name != NULL) {
```

Comment:
Per discussion, we want to see if we can simplify things and apply an invariant that this kind of "replacement" can never happen.  Unsure if we can, but it would be nice if we could.  Worth nothing that currently we don't handle the cases where we're given nothing in the new context but already have something in the current state.  Also, we can't tell if we were successfully set with null vs not set at all.